mozilla-services/socorro
# bug 1710725: stop using dotdict for raw and processed crash

This stops using `DotDict` for raw and processed crash data. When data is loaded from crash storage, it'll now be in standard Python data structures (dict, list) and needs to be accessed as such.
    
We were doing this already for the most part, so this is a minor change at this point.

The Socorro processor pipeline keeps notes related to processing as crash reports wind their way through the rules. At the end, the processor notes are added to the crash report. In this way, we have some visibility into processing.
    
Previously, `processor_meta_data` was a `DotDict` with a handful of fields. In this PR, we remove all the unused fields and then change it to a data class with a more convenient way of adding notes. Then we update all the rules and code that uses it.

To test:

1. process some crash reports and make sure there are no errors
2. reprocess a crash report and make sure the notes are extended
